src/generators/legacy-json/utils/buildSection.mjs

```diff
@@ -0,0 +1,385 @@
+import {
+  DEFAULT_EXPRESSION,
+  LEADING_HYPHEN,
+  NAME_EXPRESSION,
+  RETURN_EXPRESSION,
+  TYPE_EXPRESSION,
+} from '../constants.mjs';
+import { buildHierarchy } from './buildHierarchy.mjs';
+import parseSignature from './parseSignature.mjs';
+import { getRemarkRehype } from '../../../utils/remark.mjs';
+import { transformNodesToString } from '../../../utils/unist.mjs';
+
+const sectionTypePlurals = {
+  module: 'modules',
+  misc: 'miscs',
+  class: 'classes',
+  method: 'methods',
+  property: 'properties',
+  global: 'globals',
+  example: 'examples',
+  // Constructors should go under a class sections' signatures property
+  ctor: 'signatures',
+  classMethod: 'classMethods',
+  event: 'events',
+  var: 'vars',
+};
+
+/**
+ * @param {import('../types.d.ts').HierarchizedEntry} entry
+ * @returns {import('../types.d.ts').Meta | undefined}
+ */
+function createMeta(entry) {
+  const makeArrayIfNotAlready = val => (Array.isArray(val) ? val : [val]);
+  const { added_in, n_api_version, deprecated_in, removed_in, changes } = entry;
+  if (
+    !added_in &&
+    !n_api_version &&
+    !deprecated_in &&
+    !removed_in &&
+    changes.length < 1
+  ) {
+    return undefined;
+  }
+  return {
+    changes,
+    added: added_in ? makeArrayIfNotAlready(added_in) : undefined,
+    napiVersion: n_api_version
+      ? makeArrayIfNotAlready(n_api_version)
```

Comment:
The name `makeArrayIfNotAlready` is a bit silly just call it `enforceArray` or `toArray`

Reply:
Also IMO completely fine for these to be empty arrays instead of undefined, IMO, keeps the spec consistent and prevents downstream consumers to verify it is an array.

Allowing you to do `added: toArray(added_in ?? [])`